api/v1alpha1/validation.go

```diff
@@ -0,0 +1,41 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateLabels validates that a set of labels are correctly defined.
+func ValidateLabels(labels *map[string]string) error {
+	errs := validateLabels(labels, field.NewPath("metadata").Child("labels"))
+	return errors.Reduce(errs.ToAggregate())
+}
+
+// ValidateAnnotations validates that a set of annotations are correctly defined.
+func ValidateAnnotations(labels *map[string]string) error {
+	errs := validateLabels(labels, field.NewPath("metadata").Child("annotations"))
```

Comment:
Not all annotations are labels. 
When attempting to update a device using PUT method, an error 400 is received. This is a device that is bound to a Resource sync and has the following annotations:

```
    "annotations": {
      "device-controller/renderedVersion": "2",
      "fleet-controller/templateVersion": "2024-06-18T09:59:44Z"
    },
```

Reply:
Annotations shouldn't be settable by the user according to how we defined them

Reply:
Yes, but we're currently using PUT since that's the only request available, and the entire object needs to comply with the spec, which implies sending all mandatory fields, etc. 

So either we can blank them on the UI, or the API controller could ignore them for the validations, since anyway there are blanked out later in `NilOutManagedObjectMetaProperties`.

Also, this would not happen once we have the PATCH request available either.

Reply:
Right, the backend ignores any annotations set by the user, so the UI doesn't need to allow setting them, and the API doesn't need to validate them either I suppose.

Reply:
Annotations are not as strict as labels in k8s. For a correct k8s annotation validation, we should use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L44

Reply:
Updated the validation to use the K8s code.

Reply:
Issue fixed with the last changes, thanks